src/gui/torrentoptionsdialog.cpp

```diff
@@ -343,6 +445,37 @@ int TorrentOptionsDialog::getSeedingTime() const
     return m_ui->spinTimeLimit->value();
 }
 
+void TorrentOptionsDialog::handleCategoryChanged(const int index)
+{
+    Q_UNUSED(index);
+
+    if ((m_ui->checkAutoTMM->checkState() == Qt::Checked) && !m_ui->comboCategory->currentText().isEmpty())
+    {
+        const QString savePath = BitTorrent::Session::instance()->categorySavePath(m_ui->comboCategory->currentText());
+        m_ui->savePath->setSelectedPath(Utils::Fs::toNativePath(savePath));
+    }
+}
+
+void TorrentOptionsDialog::handleTMMChanged()
+{
+    if (!m_ui->checkAutoTMM->isChecked())
+    {
+        m_ui->labelSavePath->setEnabled(true);
+        m_ui->savePath->setEnabled(true);
+        m_ui->savePath->setSelectedPath(Utils::Fs::toNativePath(m_initialValues.savePath));
+    }
+    else
+    {
+        m_ui->labelSavePath->setEnabled(false);
+        m_ui->savePath->setEnabled(false);
+        if ((m_ui->checkAutoTMM->checkState() == Qt::Checked) && !m_ui->comboCategory->currentText().isEmpty())
```

Comment:
Why is checking for `!m_ui->comboCategory->currentText().isEmpty()` needed?

Reply:
Because it will show the default save path if it's empty.

Reply:
> Because it would show the default save path if it's empty.

And what do you expect to see in this case?
"Auto managed" torrents with "empty" category are saved under default save path, aren't they?

Reply:
Empty category also means mixed categories (in this dialog) so i missed that case. I'll pass `allSameCategory` to `handleTMMChanged()` and use that instead.

Reply:
> I'll pass `allSameCategory` to `handleTMMChanged()` and use that instead.

Doesn't seem to work with connect(). Should i make `allSameCategory` member var or maybe there is another way?

Reply:
> Should i make allSameCategory member var or maybe there is another way?

Why not?
But I would make it in a slightly different way. I would declare `m_commonCategory` member as `std::optional<QString>` and use it for both common category name and indicator of its presence. Do you understand what I mean?

Reply:
On the other hand, it's not a problem to get the name of a common category, just knowing that it exists (we still have a list of all torrents), so a Boolean member variable is enough.

Offtopic: I just generally dislike the existing names like `allSameXXX`.

Reply:
>Do you understand what I mean?

Yep, now that i've read up a little on std::optional. I think the bool is simpler though so i used that.

>Offtopic: I just generally dislike the existing names like allSameXXX.

I'm open to suggestions... that don't involve refactoring.